core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -5032,15 +5032,16 @@ private SqlDialect nonOrdinalDialect() {
    */
   @Test void testCastAsMapType() {
     sql("SELECT CAST(MAP['A', 1.0] AS MAP<VARCHAR, DOUBLE>)")
-        .ok("SELECT CAST(MAP['A', 1.0] AS MAP< VARCHAR CHARACTER SET \"ISO-8859-1\", DOUBLE >)\n"
+        .ok("SELECT CAST(MAP['A', 1.0] AS "
```

Comment:
Maybe we also need to add a test for nested Map， such as Map<VARCHAR, MAP<VARCHAR, VARCHAR>> case?

Reply:
But this method already has such a test (`AS MAP<VARCHAR ARRAY, MAP<INT, INT>>`)

Reply:
OK, I see it in the last line.